pkgs/by-name/ba/baidupcs-go/package.nix

```diff
@@ -0,0 +1,33 @@
+{
+  buildGo122Module,
+  lib,
+  fetchFromGitHub,
+  versionCheckHook,
+}:
+buildGo122Module rec {
+  pname = "baidupcs-go";
+  version = "3.9.5";
+  src = fetchFromGitHub {
+    owner = "qjfoidnh";
+    repo = "BaiduPCS-Go";
+    rev = "v${version}";
+    hash = "sha256-zNodRQzflOOB3hAeq4KbjRFlHQwknVy+4ucipUcoufY=";
+  };
+
+  vendorHash = "sha256-msTlXtidxLTe3xjxTOWCqx/epFT0XPdwGPantDJUGpc=";
+  doCheck = false;
+
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+  doInstallCheck = true;
+  versionCheckProgram = "${placeholder "out"}/bin/${meta.mainProgram}";
```

Comment:
```suggestion
  versionCheckProgram = "baidupcs-go";
```

Reply:
Does not work (from packaging `uesave` in #312965):

```
uesave> Running phase: installCheckPhase
uesave> Executing versionCheckPhase
uesave> versionCheckHook: uesave was not found, or is not an executable
```